vb-fe/src/_store/room/RoomActions.ts

```diff
@@ -1,19 +1,40 @@
-import { useSocket } from "core/socket/useSocket";
 import { Dispatch } from "redux";
-import { Room, RoomForm } from "util/Room";
-import { State } from "_store/rootReducer";
+import { push, CallHistoryMethodAction } from "connected-react-router";
+
+import { createRoom } from "services/Room";
+import { RoomForm } from "util/Room";
+import { User } from "util/User";
 import { SystemActionTypes, SystemConstants } from "_store/system/SystemTypes";
-import { RoomActionTypes } from "./RoomTypes";
+import { RoomActionTypes, RoomConstants } from "./RoomTypes";
 
 export const createRoomAction =
   (room: RoomForm) =>
-  async (dispatch: Dispatch<RoomActionTypes | SystemActionTypes>, getState: () => State): Promise<void> => {
-    const { socket } = getState().system;
-
+  async (dispatch: Dispatch<RoomActionTypes | SystemActionTypes | CallHistoryMethodAction>): Promise<void> => {
     dispatch({
       type: SystemConstants.LOADING,
-      payload: true,
+      payload: false,
```

Comment:
Shouldn't loading be in effect while we wait for the API call to fail or succeed?

Reply:
oh yeah i totalyl forgot